### DISCUSSION TITLE
Documentation on registry_package event

### DISCUSSION THREAD

**Discussion by riker09** [community/community#26523] [Actions]
<p>It seems the documention for the registry_package event is missing.</p>
<p>People are using it already (see <a href="https://github.community/t/trigger-workflow-on-registry-package-event/16437">here</a> and <a href="https://github.community/t/registry-package-trigger-with-docker-image-in-github-package-registry/16673">here</a>), so it must have been documented at some point.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/riker09">@riker09</a> ,</p>
<p>Thanks for your feedback!</p>
<p>I have submitted a query for the event ‘registry_package’ doc internally, will update once there’s a response.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/riker09">@riker09</a> ,</p>
<p>There is a ticket about this event in doc improvement on tracking, please check the screenshot below. </p>
<p>Will update once it’s completed. Thanks!</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4b9f869231d6584ade3c1641006e9332c1c992ed.png" data-download-href="/uploads/short-url/aMZAJnOlKPuJIRZYlkhLOa7VPIh.png?dl=1" title="docfix.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181098143-fec5323c-13dd-49f5-a032-6851e5aac9c7.png" alt="docfix.png" data-base62-sha1="aMZAJnOlKPuJIRZYlkhLOa7VPIh" width="690" height="371" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4b9f869231d6584ade3c1641006e9332c1c992ed_2_690x371.png, https://aws1.discourse-cdn.com/github/original/2X/4/4b9f869231d6584ade3c1641006e9332c1c992ed.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/4b9f869231d6584ade3c1641006e9332c1c992ed.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4b9f869231d6584ade3c1641006e9332c1c992ed_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">docfix.png</span><span class="informations">773×416 59.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by riker09** [comment]
<p>FWIW, here’s the link to the documentation: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#registry-package-event-registry_package" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#registry-package-event-registry_package</a></p>
<p>And since the internet is a volatile thing, here’s the documentation as the time of writing (sorry for posting it as an image):</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/31ab8604ce2ccf770df3c6a6d4a018c45c4ece69.png" data-download-href="/uploads/short-url/75oSM5ObvQIpXJncfGRRHFMCqxj.png?dl=1" title="Screenshot_2020-02-05 Events that trigger workflows - GitHub Help.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/568497/181098144-f3b7aef4-0b8e-430e-98d6-3bfea9f3a833.png" alt="Screenshot_2020-02-05 Events that trigger workflows - GitHub Help.png" data-base62-sha1="75oSM5ObvQIpXJncfGRRHFMCqxj" width="690" height="450" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/31ab8604ce2ccf770df3c6a6d4a018c45c4ece69_2_690x450.png, https://aws1.discourse-cdn.com/github/original/2X/3/31ab8604ce2ccf770df3c6a6d4a018c45c4ece69.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/31ab8604ce2ccf770df3c6a6d4a018c45c4ece69.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/31ab8604ce2ccf770df3c6a6d4a018c45c4ece69_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_2020-02-05 Events that trigger workflows - GitHub Help.png</span><span class="informations">739×483 51.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><em>PLEASE NOTE: Information gets outdated very quickly, so if you stumble upon this in the future make sure this documentation is still valid.</em></p>

**Comment by antl3x** [comment]
<p>This is still valid?</p>
<p>Seems that the Github Event Webhhok API was updated and  <strong>registry_package</strong> was renamed to  <strong>package</strong>.</p>
<p><a href="https://github.com/octokit/webhooks/releases/tag/v3.0.0" rel="nofollow noopener">https://github.com/octokit/webhooks/releases/tag/v3.0.0</a></p>

**Comment by riker09** [comment]
<p>If there has been a rename it has not (yet) made it into the documentation. Here is a screenshot from today (Wed. 26th, Feb. 2020):</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/c315efe74b48c8d991ee0a96cfeff4cff47928bc.png" data-download-href="/uploads/short-url/rPO6Cz4aScc4FuLPx6vdNH9bjsg.png?dl=1" title="Bildschirmfoto von 2020-02-26 11-35-43.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/568497/181098148-c0c41ecb-ec79-499d-bca2-dff515c01267.png" alt="Bildschirmfoto von 2020-02-26 11-35-43.png" data-base62-sha1="rPO6Cz4aScc4FuLPx6vdNH9bjsg" width="690" height="418" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/c315efe74b48c8d991ee0a96cfeff4cff47928bc_2_690x418.png, https://aws1.discourse-cdn.com/github/original/1X/c315efe74b48c8d991ee0a96cfeff4cff47928bc.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/c315efe74b48c8d991ee0a96cfeff4cff47928bc.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/c315efe74b48c8d991ee0a96cfeff4cff47928bc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Bildschirmfoto von 2020-02-26 11-35-43.png</span><span class="informations">812×492 51.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Since there is documentation available I guess this topic can be marked as solved. However, I still believe that the feature still needs more attention (like a package name filter) but that is a different story.</p>
<p>[EDIT]</p>
<p>Wait, I already accepted a reply as solution, so nevermind. :slight_smile:</p>

**Comment by jayarjo** [comment]
<p>_registry_package _wasn’t working for me. I tried to rename <em>registry_package</em> to <em>package</em>, but I get the following error:</p>
<p><img src="https://user-images.githubusercontent.com/150075/181098149-e349b7a9-8720-48ec-aa5f-6f6282731fd5.png" alt="Screenshot from 2020-04-04 20-38-14.png" data-base62-sha1="cZibMb6S74G56boXUuGX7TgZH4U" width="523" height="236"></p>

**Comment by jayarjo** [comment]
<p>If it wasn’t documented, how exactly people found it to use it?</p>

**Comment by riker09** [comment]
<p>It was documented at <em>some</em> point, and then it got removed later on. The documentation is updated to the latest changes of GitHub Actions. Sometimes documentation and actual behavior diverge. I would love to see this issue addressed, by having the documentation to be in sync with the released features. It would be helpful to have a documentation changelog as well. And GitHub Actions (Core) should have version tags which then could be referenced in the documentation. I have not spent any research time whatsoever to see if this is an actual feature request somewhere else.</p>

**Comment by riker09** [comment]
<p>Yeah, I get the same error now. <code>registry_package</code> is not working (anymore?), the workflow does not even show up in the list of workflows of the repository. And renaming the event to <code>package</code> results in the same error “Invalid workflow file”.</p>

**Comment by 64J0** [comment]
I would like to know how to get the package name, so I can use it later in the workflow.
